platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -260,6 +267,7 @@ public Property<InsecureProtocolOption> getInsecureProtocol() {
     @Option(option = "comments", description = "Include clarifying comments in files.")
     public abstract Property<Boolean> getComments();
 
+    @ToBeKeptEagerProperty(because = "Injected service")
     public ProjectLayoutSetupRegistry getProjectLayoutRegistry() {
```

Comment:
❓ Why isn't this marked with `@Inject` then?

We should also not require an annotation on anything that has `@Inject`.

Reply:
This has a public setter too. Not sure why it's used though. But it's a breaking change if I remove it and add just `@Inject` annotation. So I can open an issue for that.